elki/src/main/java/de/lmu/ifi/dbs/elki/algorithm/associationrulemining/AssociationRule.java

```diff
@@ -0,0 +1,160 @@
+import de.lmu.ifi.dbs.elki.algorithm.itemsetmining.OneItemset;
+import de.lmu.ifi.dbs.elki.algorithm.itemsetmining.SparseItemset;
+import de.lmu.ifi.dbs.elki.data.BitVector;
+import de.lmu.ifi.dbs.elki.data.type.VectorFieldTypeInformation;
+
+/**
+ * Association Rule
+ * 
+ * @author Frederic Sautter
+ *
+ */
+public class AssociationRule implements Comparable<AssociationRule> {
+  
+  /**
+   * Consequent itemset
+   */
+  private Itemset consequent;
+  
+  /**
+   * Union of consequent and consequent
+   */
+  private Itemset union;
+  
+  /**
+   * Support of the antecedent
+   */
+  private int anteSupport;
+  
+  /**
+   * Measure of the Rule.
+   */
+  private double measure;
+
+  /**
+   * Constructor.
+   *
+   * @param union Union of consequent and consequent
+   * @param consequent Consequent of the rule
+   * @param anteSupport Support of the antecedent
+   * @param measure Value of the interest measure
+   */
+  public AssociationRule(Itemset union, Itemset consequent, int anteSupport, double measure) {
+    // TODO Auto-generated constructor stub
+    this.union = union;
+    this.consequent = consequent;
+    this.anteSupport = anteSupport;
+    this.measure = measure;
+  }
+  
+  
+  
+  /**
+   * Get the consequent of the rule.
+   * 
+   * @return consequent
+   */
+  public Itemset getConsequent() {
+    return this.consequent;
+  }
+  
+  /**
+   * Get the union of consequent and consequent of the rule.
+   * 
+   * @return union
+   */
+  public Itemset getUnion() {
+    return this.union;
+  }
+  
+  /**
+   * Get the antecedent support.
+   * 
+   * @return confidence
+   */
+  public int getAnteSupport() {
+    return this.anteSupport;
+  }
+  
+  /**
+   * Get the rule measure.
+   * 
+   * @return measure
+   */
+  public double getMeasure() {
+    return this.measure;
+  }
+  
+  /**
+   * Get the antecedent of the rule.
+   * 
+   * @return antecedent
+   */
+  public Itemset getAntecedent() {
+    int[] itemset = union.toSparseRep();
+    int[] con = consequent.toSparseRep();
+    int[] antecedent = new int[itemset.length - con.length];
+    int i = 0;
+    int index = 0;
+    for(int item : itemset) {
+      if(item != con[i]) {
+          antecedent[index] = item;
+          index++;
+      } else if(i != con.length - 1) {
+          i++;
+      }
+    }
+    if(antecedent.length == 1) {
+      return new OneItemset(antecedent[0], anteSupport);
+    } else {
+      return new SparseItemset(antecedent, anteSupport);
+    }
+  }
+
+  @Override
+  public int compareTo(AssociationRule o) {
+    // TODO Auto-generated method stub
+    return this.union.compareTo(o.union);
+  }
+  
+  /**
+   * Append to a string buffer.
+   * 
+   * @param buf Buffer
+   * @param meta Relation metadata (for labels)
+   * @return String buffer for chaining.
+   */
+  public StringBuilder appendTo(StringBuilder buf, VectorFieldTypeInformation<BitVector> meta){
+    buf.append(this.getAntecedent().appendTo(new StringBuilder(), meta));
```

Comment:
Why not simply `getAntecedent().appendTo(buf, meta);`? Why a new `StringBuilder`?